### PR TITLE
cb3 syntax, use more global pins

### PR DESCRIPTION
**Pull request by dougalsutherland**
Modernize the `meta.yaml` a little bit, including pulling in more pins automatically. Only real effect on the output should be that it builds with an older `numpy` when possible.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -40,19 +38,13 @@ requirements:
...
-    - vc 14  # [win]
-    - curl
-    - zlib
+    - {{ pin_compatible('numpy') }}
+    - {{ pin_compatible('jsoncpp', max_pin='x.x.x') }}
+    - {{ pin_compatible('postgresql', max_pin='x.x') }}
+    - {{ pin_compatible('laz-perf', max_pin='x.x') }}
+    - {{ pin_compatible('nitro', max_pin='x.x') }}
+    - {{ pin_compatible('hexer', max_pin='x.x') }}
+    - {{ pin_compatible('laszip', max_pin='x.x') }}
```
Comment (bgruening) [inline recipe/meta.yaml]: @dougalsutherland any reason to leave zlib out? I does not have a run_exports yet, isn't it?

Comment (dougalsutherland) [reply recipe/meta.yaml]: There's a `pin_run_as_build` in `conda-forge-pinning`:

https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/793426c3b16a049fc0bdaba1a731103b9548c6f7/recipe/conda_build_config.yaml#L199-L200

Comment (bgruening) [reply recipe/meta.yaml]: Ah cool, missed that. Thanks.

**Comment by dougalsutherland** [comment]
Merging this too, because it'll help a tiny bit with https://github.com/conda-forge/staged-recipes/pull/5724.

**Pull request merged**